--- swig/python/setup.py.orig	2022-08-04 13:00:57.659818933 +1000
+++ swig/python/setup.py	2022-08-04 13:04:30.130263637 +1000
@@ -242,9 +242,9 @@
         elif isinstance(self.library_dirs, str) and sys.platform == 'darwin':
             self.library_dirs += ':' + ':'.join(library_dirs)
         if self.libraries is None:
-            if self.get_compiler() == 'msvc':
-                libraries.remove('gdal')
-                libraries.append('gdal_i')
+            #if self.get_compiler() == 'msvc':
+            #    libraries.remove('gdal')
+            #    libraries.append('gdal_i')
             self.libraries = libraries
 
         build_ext.finalize_options(self)
